rust/xaynet-server/src/storage/store.rs

```diff
@@ -0,0 +1,159 @@
+//! A generic store.
+
+use async_trait::async_trait;
+
+use crate::{
+    state_machine::coordinator::CoordinatorState,
+    storage::{
+        CoordinatorStorage,
+        LocalSeedDictAdd,
+        MaskScoreIncr,
+        ModelStorage,
+        StorageResult,
+        SumPartAdd,
+    },
+};
+use xaynet_core::{
+    common::RoundSeed,
+    mask::{MaskObject, Model},
+    LocalSeedDict,
+    SeedDict,
+    SumDict,
+    SumParticipantEphemeralPublicKey,
+    SumParticipantPublicKey,
+    UpdateParticipantPublicKey,
+};
+
+#[derive(Clone)]
+/// A generic store.
+pub struct Store<C, M>
+where
+    C: CoordinatorStorage,
+    M: ModelStorage,
+{
+    /// A coordinator store.
+    coordinator: C,
+    /// A model store.
+    model: M,
+}
+
+impl<C, M> Store<C, M>
+where
+    C: CoordinatorStorage,
+    M: ModelStorage,
+{
+    /// Creates a new [`Store`].
+    pub fn new(coordinator: C, model: M) -> Self {
+        Self { coordinator, model }
+    }
+}
+
+#[async_trait]
+impl<C, M> CoordinatorStorage for Store<C, M>
+where
+    C: CoordinatorStorage,
+    M: ModelStorage,
+{
+    /// See [`CoordinatorStorage::set_coordinator_state`].
```

Comment:
If I'm not mistaken the doc is inherited from the trait definition, so this should not be needed.

Reply:
good point 👍 I removed it